datalad_next/runners/data_processors.py

```diff
@@ -0,0 +1,316 @@
+"""
+This module contains a generator that reads data-chunks from an iterable,
```

Comment:
I assume that generator is `ProcessorPipeline`?

From my POV it would make sense to cleanly split the pipelining aspect from the data processor implementations, and place them in different files.

Reply:
Yes, it makes sense. I wanted to prevent too many modules that we have to import from. We might still put the implementations into different files and import the processors into the name space of the pipelining code, so that they could be imported from there!?

Reply:
From the POV of the `datalad_next` code base, module number minimization is not a goal. If some things are "standard" components, they can be propagated to the mini-package root. However, a new feature is hardly a "standard" instantaneously.

Reply:
I will split them up in a new commit.